File: panacea.rb
######################
##### BootCoder ######
######################
## Hunter T. Chapman #
######################
##     PANACEA      ##
######################



# INPUT SAMPLE

# 64 6e 78 | 100101100 11110
# 5e 7d 59 | 1101100 10010101 1100111
# 93 75 | 1000111 1011010 1100010

# OUTPUT SAMPLE

# True
# True
# False


##########################
### Refactored Solution ##
##########################

# def calculate_totals(hexs, binarys)
#   hex_total = hexs.reduce(0) {|sum,num| sum += num.to_i(16)}
#   binary_total = binarys.reduce(0) {|sum,num| sum += num.to_i(2)}
#   puts hex_total <= binary_total ? "True" : "False"
# end

# # File.open('input_sample.txt').each_line do |line|
# File.open(ARGV[0]).each_line do |line|
#   line = line.chomp.split("|")
#   hexs = line[0].split(" ")
#   binarys = line[1].split(" ")
#   calculate_totals(hexs, binarys)
# end


#######################
### Initial Solution ##
#######################

nums = []
# require 'awesome_print'
File.open('input_sample.txt').each_line do |line|
# File.open(ARGV[0]).each_line do |line|
  line = line.chomp.split("|")
  hexs = line[0].split(" ")
  binarys = line[1].split(" ")
  nums << {hexs: hexs, binarys: binarys}
end

def calculate_totals(nums)
  results = []
  nums.each do |line|

    hex_total = 0
    binary_total = 0

    line[:hexs].each do |hex|
      hex_total += hex.to_i(16)
      p hex.to_i(16)
    end
    puts "----"
    line[:binarys].each do |binary|
      binary_total += binary.to_i(2)
      p binary.to_i(2)
    end
    p "VIRUS_____total: #{hex_total}"
    p "ANTIVIRUS_total: #{binary_total}"
    result = hex_total <= binary_total ? true : false
    results << result
  end
  results
end



calculate_totals(nums).each do |result|
  p result
end

File: panacea.js

var fs  = require("fs");
fs.readFileSync('input_sample.txt').toString().split('\n').forEach(function (line) {
// fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line != "") {
      var splitLine = line.split("|")
      var hexNums = splitLine[0].split(" ")
      var binNums = splitLine[1].split(" ")
      hexNums.pop()
      binNums.shift()
      // console.log(hexNums)
      // console.log(binNums)
      var binNum = 0;
      var hexNum = 0;
      for(var i = 0; i < hexNums.length; i += 1){
        hexNum += parseInt(hexNums[i],16);
      }
      for(var i = 0; i < binNums.length; i += 1){
        binNum += parseInt(binNums[i],2);
      }

      if(hexNum <= binNum){
        console.log("True");
      }else{
        console.log("False");
      }
      // var hexSum = hexNums.reduce(function(a,b){ return parseInt(a,16) + parseInt(b,16)},0)
      // var binSum = binNums.reduce(function(a,b){ return parseInt(a,2) + parseInt(b,2)},0)
      // console.log('hexNums: '+hexSum);
      // console.log('binaryNums: '+binSum);
    }
});

// var sum = a.reduce(function(a, b) { return a + b; }, 0);

